cocotb/queue.py

```diff
@@ -35,6 +41,9 @@ def __init__(self, maxsize: int = 0):
     def _init(self, maxsize):
         self._queue = collections.deque()
 
+    def _peek(self):
+        return self._queue[0]
```

Comment:
This raises the wrong exception when the queue is empty.

Reply:
Let's reject this pull request.  I'll put in another one with only the exceptions.